### ISSUE TITLE
Check for absence of header

### ISSUE THREAD

**Issue by acanda** [mock-server/mockserver#867]
**Describe the feature request**
I would like to verify that a specific header (e.g. Authorization) is not present in the request.

**What you are trying to do**
I am trying to verify that a specific header (e.g. Authorization) is not present in the request. With version 5.11.1 of mockserver-netty I did not find a way to do that.

**The solution you'd like**
I would like to write something along the line of
```
clientAndServer.verify(request().withoutHeader("Authorization"));
```
that verifies that the request did not contain the header "Authorization". This should fail if there is a header "Authorization" without a value. It should only succeed if the header "Authorization" is completely absent.

**Describe alternatives you've considered**
I have found issue #159 from five years ago that claims that the following should work:
```
withHeader(NottableString.not("Authorization"), NottableString.string(".*"))
```
But it does not work (anymore) and I also do not think it should work as this should match any header that is not "Authorization" and has a value matching ".*". So this should match e.g. `Content-Type: application/json`.

I also tried the following, but none of them work:
```
withHeader("Authorization")  // this is the same as withHeader("Authorization", ".*")
withHeader("Authorization", "^$")  // this only matches if the header is present with an empty value
withHeader(NottableString.not("Authorization"))  // this matches any header that is not "Authorization"
withHeader(NottableString.string("Authorization"), NottableString.not(".*"))  // this seems to never match anything
```

If I understand the logic behind `withHeader(...)` and `NottableString` correctly, then there is no way to express the absence of a header with any combination of them.

**Comment by jamesdbloom** [comment]
I've added a test shows that it is possible to write an expectation that only matches when a header is not present, the code can be seen here: https://github.com/mock-server/mockserver/blob/fd0fe7e84eb4053ad793111d3d36c610da6a6992/mockserver-integration-testing/src/main/java/org/mockserver/testing/integration/mock/AbstractExtendedMockingIntegrationTest.java#L432

As this test demonstrates this is possible I'm closing this issue.

**Comment by davidwcarlson** [comment]
There was a similar issue #821 in versions 5.11.0 and 5.11.1 which was fixed in 5.11.2
